File: contests/abc219/abc219_c/main.cc
#include <bits/stdc++.h>

#include "atcoder.h"

void Main() {
  strings(x);
  V<int> ord(26);
  rep(i, 26) ord[x[i] - 'a'] = i;
  ints(n);
  V<string> s(n);
  cin >> s;
  V<pair<string, string>> ans;
  rep(i, n) {
    string t = s[i];
    each(e, t) e = 'a' + ord[e - 'a'];
    ans.eb(t, s[i]);
  }
  sort(all(ans));
  each(_, e, ans) wt(e);
}

File: contests/abc219/abc219_c/main.go
package main

import (
	"sort"

	"github.com/kkishi/hytk/rd"
	"github.com/kkishi/hytk/wt"
)

func main() {
	x := rd.String()
	var ord [26]int
	for i := range ord {
		ord[x[i]-'a'] = i
	}
	n := rd.Int()
	s := rd.Strings(n)
	type P struct {
		s, t string
	}
	var ans []P
	for i := range s {
		var t []byte
		for _, e := range s[i] {
			t = append(t, byte('a'+ord[e-'a']))
		}
		ans = append(ans, P{
			s: s[i],
			t: string(t),
		})
	}
	sort.Slice(ans, func(i, j int) bool {
		return ans[i].t < ans[j].t
	})
	for _, e := range ans {
		wt.W(e.s)
	}
}
